File: 7._Date_servers/node/main.js
import express from 'express';
import path from 'path';

const app = express();
app.use(express.static(path.resolve('./public')));

app.get("/timestamp", (req, res) => {
    res.send({"timestamp": new Date().toISOString()});
})

app.get("/pythontimestamp", async (req, res) => {
    const pythonResponse = await fetch("http://127.0.0.1:8000/timestamp")
    const pythonMessage = await pythonResponse.json();
    //parse into UTC datetime
    res.send({"pythonTimestamp": new Date(pythonMessage['timestamp'])});
})

app.listen(3000, (error) => {
    console.log("Server is running on", 3000);
})

File: 7._Date_servers/python/main.py
from fastapi import FastAPI
import requests
import datetime

app = FastAPI()
@app.get("/nodetimestamp")
def _():
    response = requests.get("http://localhost:3000/timestamp")
    #strip last Z to parse to UTC datetime
    return {'nodeTimestamp': datetime.datetime.fromisoformat(response.json()['timestamp'][:-1])}

@app.get("/timestamp")
def _():
    return {"timestamp": datetime.datetime.utcnow().isoformat()+"Z"} #add military timezone suffix

